src/main/java/seedu/ddd/model/AddressBook.java

```diff
@@ -169,27 +173,48 @@ public void setEvent(Event target, Event editedContact) {
         //events.setEvents(target, editedContact);
     }
 
+    /**
+     * Fetches an {@code Event} from the address book by ID.
+     * If event has not been created or does not exist, a null object will be returned.
+     */
+    public Event getEvent(Id eventId) {
+        return events.get(eventId);
+    }
+
     /**
      * Removes {@code key} from this {@code AddressBook}.
      * {@code key} must exist in the address book.
+     * @throws CommandException when there are event(s) that are dependent on the contact
      */
-    public void removeContact(Contact key) {
-        contacts.remove(key);
-        List<Event> contactEvents = key.getEvents().stream().toList();
-        for (Event event: contactEvents) {
-            events.get(event.getEventId()).removeContact(key);
+    public void deleteContact(Contact key) throws CommandException {
+        if (key instanceof Client) {
+            long numEvents = key.getEvents().stream()
+                    .map(Event::getClients)
+                    .filter(clientList -> clientList.size() == 1)
+                    .count();
+            if (numEvents > 0) {
+                throw new CommandException(String.format(MESSAGE_DEPENDENT_EVENT, key.getEvents().size()));
```

Comment:
Can we change `MESSAGE_DEPENDENT_EVENT` to show the events that depend on the client instead of just the number of dependent events?